### DISCUSSION TITLE
VW install on YunoHost

### DISCUSSION THREAD

**Discussion by mychristopherta** [dani-garcia/vaultwarden#2175] [General]
Fresh install of YunoHost, version 4.3.4.2. Installed Vaultwarden as first app. 
Received this error: info: default toolchain set to 'nightly-x86_64-unknown-linux-gnu'
System just hangs there indefinitely.  I can only recover by restarting the server. Once I do, accessing /vaultwarden install nets Bad Gateway. Is this a bad package?
Any assistance here is much appreciated!

**Comment by RealOrangeOne** [comment]
Looking at how the yunohost configuration works, it compiles vaultwarden from source. This can take quite a while. 

> info: default toolchain set to 'nightly-x86_64-unknown-linux-gnu'

This is not an error, this is expected, and happens during the compilation.

My suspicion is that what this "hang" is is actually your system compiling Vaultwarden. Have a look at something like `htop` and see what the system is doing during that time. Restarting will likely result in a broken vaultwarden install, as the compilation won't have finished, hence the "Bad gateway".
